pkg/segment/sortindex/sortindex.go

```diff
@@ -547,3 +559,47 @@ func IsEOF(checkpoint *Checkpoint) bool {
 
 	return checkpoint.eof
 }
+
+func SetSortColumns(columnNames []string) error {
+	if columnNames == nil {
+		columnNames = make([]string, 0)
+	}
```

Comment:
nit: you don't really need this. I think you're doing this to avoid an issue when iterating over a nil slice, but Go already handles that